File: src/test/java/voidchess/helper/ExtendedMoveTest.java
package voidchess.helper;

import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;
import voidchess.figures.Pawn;

import static org.testng.Assert.assertFalse;
import static org.testng.Assert.assertTrue;

/**
 * @author stephan
 */
public class ExtendedMoveTest {
    private ExtendedMove extendedMove;

    @BeforeMethod
    protected void setUp() {
        extendedMove =
                new ExtendedMove(
                        Move.get(
                                Position.byCode("a1"),
                                Position.byCode("d1")
                        ),
                        null,
                        null,
                        true,
                        false,
                        false,
                        false);
    }

    @Test
    public void testFrom() {
        assertTrue(extendedMove.getMove().from.equalsPosition(Position.byCode("a1")));
    }

    @Test
    public void testTo() {
        assertTrue(extendedMove.getMove().to.equalsPosition(Position.byCode("d1")));
    }

    @Test
    public void testGetFigure() {
        assertTrue(extendedMove.getFigureTaken() == null);
    }

    @Test
    public void testGetColorOfMove() {
        assertTrue(extendedMove.getColorOfMove());
    }

    @Test
    public void testIsCastling() {
        assertFalse(extendedMove.isCastling());
    }

    @Test
    public void testIsEnpassent() {
        assertFalse(extendedMove.isEnpassent());
    }

    @Test
    public void testHasHitFigure() {
        assertFalse(extendedMove.wasFigureTaken());
        ExtendedMove hitPawn =
                new ExtendedMove(
                        Move.get(
                                Position.byCode("b5"),
                                Position.byCode("a6")
                        ),
                        null,
                        new Pawn(false, Position.byCode("a5"), true),
                        true,
                        false,
                        true,
                        false
                );
        assertTrue(hitPawn.wasFigureTaken());
        ExtendedMove hitFigure =
                new ExtendedMove(
                        Move.get(
                                Position.byCode("b5"),
                                Position.byCode("a6")
                        ),
                        new Pawn(false, Position.byCode("a6"), false),
                        null,
                        true,
                        false,
                        true,
                        false);
        assertTrue(hitFigure.wasFigureTaken());
    }
}

File: module-ui/src/test/kotlin/voidchess/board/move/ExtendedMoveTest.kt
package voidchess.board.move

import org.testng.annotations.BeforeMethod
import org.testng.annotations.Test
import voidchess.figures.Pawn

import org.testng.Assert.assertFalse
import org.testng.Assert.assertTrue
import voidchess.board.move.ExtendedMove
import voidchess.common.board.move.Move
import voidchess.common.board.move.Position


class ExtendedMoveTest {
    private var extendedMove: ExtendedMove? = null

    @BeforeMethod
    private fun setUp() {
        extendedMove = ExtendedMove(
            move = Move[Position.byCode(
                "a1"
            ), Position.byCode("d1")],
            figureTaken = null, enpassantPawnOrCastlingRook = null,
            colorOfMove = true,
            isCastling = false,
            isEnPassant = false,
            isPawnTransformation = false
        )
    }

    @Test
    fun testFrom() {
        assertTrue(extendedMove!!.move.from.equalsPosition(
            Position.byCode(
                "a1"
            )
        ))
    }

    @Test
    fun testTo() {
        assertTrue(extendedMove!!.move.to.equalsPosition(
            Position.byCode(
                "d1"
            )
        ))
    }

    @Test
    fun testGetFigure() {
        assertTrue(extendedMove!!.figureTaken == null)
    }

    @Test
    fun testGetColorOfMove() {
        assertTrue(extendedMove!!.colorOfMove)
    }

    @Test
    fun testIsCastling() {
        assertFalse(extendedMove!!.isCastling)
    }

    @Test
    fun testIsEnpassent() {
        assertFalse(extendedMove!!.isEnPassant)
    }

    @Test
    fun testHasHitFigure() {
        assertFalse(extendedMove!!.wasFigureTaken)
        val hitPawn = ExtendedMove(
            move = Move[Position.byCode(
                "b5"
            ), Position.byCode("a6")], figureTaken = null,
            enpassantPawnOrCastlingRook = Pawn(
                false,
                Position.byCode("a5"),
                true
            ),
            colorOfMove = true,
            isCastling = false,
            isEnPassant = true,
            isPawnTransformation = false
        )
        assertTrue(hitPawn.wasFigureTaken)
        val hitFigure = ExtendedMove(
            move = Move[Position.byCode(
                "b5"
            ), Position.byCode("a6")],
            figureTaken = Pawn(
                false,
                Position.byCode("a6"),
                false
            ), enpassantPawnOrCastlingRook = null,
            colorOfMove = true,
            isCastling = false,
            isEnPassant = true,
            isPawnTransformation = false
        )
        assertTrue(hitFigure.wasFigureTaken)
    }
}
